apps/developer-hub/src/components/EntropyTable/index.tsx

```diff
@@ -120,3 +129,80 @@ const getEntropyDeployments = async (
   const url = isMainnet ? FORTUNA_API_URLS.mainnet : FORTUNA_API_URLS.testnet;
   return await fetchEntropyDeployments(url);
 };
+
+function useEntropyFees(
+  chains: Record<string, EntropyDeployment>,
+): Record<string, bigint | undefined> {
+  const [feesByChain, setFeesByChain] = useState<
+    Record<string, bigint | undefined>
+  >({});
+  const clientsRef = useRef<Map<string, PublicClient>>(new Map());
+
+  function getClient(rpc: string): PublicClient {
+    const cached = clientsRef.current.get(rpc);
+    if (cached) return cached;
+    const client = createPublicClient({
+      transport: http(rpc, { timeout: 10_000, retryCount: 1 }),
+    });
+    clientsRef.current.set(rpc, client);
+    return client;
+  }
+
+  useEffect(() => {
+    const entries = Object.entries(chains);
+    if (entries.length === 0) return;
+
+    let isCancelled = false;
+
+    async function loadInBatches() {
+      const batchSize = 5;
+      for (let i = 0; i < entries.length; i += batchSize) {
+        if (isCancelled) return;
+        const batch = entries.slice(i, i + batchSize);
+        const results = await Promise.allSettled(
+          batch.map(async ([name, d]) => {
```

Comment:
We should avoid using single-letter variable names, it makes the code unnecessarily hard to understand